packages/config/src/api/site_info.ts

```diff
@@ -95,19 +126,25 @@ const getAddons = async function (api: NetlifyAPI, siteId: string) {
     return Array.isArray(addons) ? addons : []
   } catch (error) {
     throwUserError(`Failed retrieving addons for site ${siteId}: ${error.message}. ${ERROR_CALL_TO_ACTION}`)
+    // Added to satisfy TypeScript only an array can be returned
```

Comment:
Instead, I would adapt the type signature of `throwUserError` so that it returns the type `never` because then it will be treated like a `throw new Error`, and no fake return is needed. (as it should never reach that return statement)